lib/models/IrcClientConfig.js

```diff
@@ -37,12 +44,46 @@ class IrcClientConfig {
         return this._config.username;
     }
 
+    setPrivateKey(privateKey) {
+        if (this._privateKey) {
+            throw new Error("Cannot override previously set privateKey!");
+        }
+        if (typeof privateKey !== 'string') {
+            let actual = typeof privateKey;
+            throw new Error("Private key must be an RSA PEM-formatted string, not " + actual);
```

Comment:
This all feels like needless boilerplate. We load the PEM key from file in `DataStore`, why aren't we just checking the types there? Why do we pass the private key through as a member variable to `IrcClientConfig`? Why is it even an option to store passwords unencrypted? It means you get these scenarios where you have no idea what `_config.password` is (unencrypted or encrypted) and you just make more work for yourself.